modules/service/src/main/scala/ch/epfl/bluebrain/nexus/iam/service/Main.scala

```diff
@@ -66,30 +68,37 @@ object Main {
 
     // cluster join hook
     cluster.registerOnMemberUp({
+      implicit val oidcConfig = appConfig.oidc
+      implicit val keys: Map[TokenId, PublicKey] =
+        oidcConfig.providers.foldLeft(Map.empty[TokenId, PublicKey]) { (acc, currentConfig) =>
+          logger.info(
+            s"Retrieving Json Web Key for issuer '${currentConfig.issuer}' from endpoint '${currentConfig.jwkCert}'")
+          acc ++ Await.result(JwkClient(currentConfig), 10.seconds)
```

Comment:
I don't think we should be loading `JWK`s at startup, at least not in a synchronous manner.
The way it's done now, we're depending on an external service to be up and running when we're starting the app, which means we cannot do a restart/deploy when external OIDC provider is down.

Reply:
The whole IAM logic relays on JWT and in order to have JWT you need to have the public keys. Either we copy the json content from it into configuration and then we load it safely or we need to rely on it in startup or somewhere else. In any case, we have to rely on the `OIDC provider`, because that's where the subsequent calls happen.